filebeat.tests.system.test_crawler.Test.test_file_no_permission is flaky

## Flaky Test

* **Test Name:** filebeat.tests.system.test_crawler.Test.test_file_no_permission
* **Link:** [Link to file/line number in github.](https://github.com/elastic/beats/blob/main/filebeat/tests/system/test_crawler.py#L760)
* **Branch:** main
* **Artifact Link:** https://beats-ci.elastic.co/job/Beats/job/beats/job/main/179/
* **Notes:** Frequently fails on Windows 8, rarely or never observed on other platforms.

### Stack Trace

```python
..\libbeat\tests\system\beat\beat.py:396: in setUp
    shutil.rmtree(self.working_dir)
C:\Python38\lib\shutil.py:737: in rmtree
    return _rmtree_unsafe(path, onerror)
C:\Python38\lib\shutil.py:610: in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
C:\Python38\lib\shutil.py:615: in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

path = 'C:\\Users\\jenkins\\workspace\\main-179-b64fe779-3b57-460f-ad15-9c22ffe1de2c\\src\\github.com\\elastic\\beats\\filebeat\\build\\system-tests\\run\\test_crawler.Test.test_file_no_permission\\log'
onerror = <function rmtree.<locals>.onerror at 0x0000008F59989DC0>

    def _rmtree_unsafe(path, onerror):
        try:
            with os.scandir(path) as scandir_it:
                entries = list(scandir_it)
        except OSError:
            onerror(os.scandir, path, sys.exc_info())
            entries = []
        for entry in entries:
            fullname = entry.path
            if _rmtree_isdir(entry):
                try:
                    if entry.is_symlink():
                        # This can only happen if someone replaces
                        # a directory with a symlink after the call to
                        # os.scandir or entry.is_dir above.
                        raise OSError("Cannot call rmtree on a symbolic link")
                except OSError:
                    onerror(os.path.islink, fullname, sys.exc_info())
                    continue
                _rmtree_unsafe(fullname, onerror)
            else:
                try:
>                   os.unlink(fullname)
E                   PermissionError: [WinError 5] Access is denied: 'C:\\Users\\jenkins\\workspace\\main-179-b64fe779-3b57-460f-ad15-9c22ffe1de2c\\src\\github.com\\elastic\\beats\\filebeat\\build\\system-tests\\run\\test_crawler.Test.test_file_no_permission\\log\\test.log'

C:\Python38\lib\shutil.py:613: PermissionError
```

--- a/filebeat/tests/system/test_crawler.py
+++ b/filebeat/tests/system/test_crawler.py
@@ -761,6 +761,32 @@ class Test(BaseTest):
         """
         Checks that filebeat handles files without reading permission well
         """
+        if os.name == "nt":
+            # Currently skipping this test on windows as it requires `pip install win32api`
+            # which seems to have windows only dependencies.
+            # To solve this problem a requirements_windows.txt could be introduced which would
+            # then only be used on Windows.
+            #
+            # Below is some additional code to give some indication on how the implementation
+            # to remove permissions on Windows (where os.chmod isn't enough) could look like:
+            #
+            # from win32 import win32api
+            # import win32security
+            # import ntsecuritycon as con
+
+            # user, domain, type = win32security.LookupAccountName(
+            #     "", win32api.GetUserName())
+            # sd = win32security.GetFileSecurity(
+            #     testfile, win32security.DACL_SECURITY_INFORMATION)
+
+            # dacl = win32security.ACL()
+            # # Remove all access rights
+            # dacl.AddAccessAllowedAce(win32security.ACL_REVISION, 0, user)
+
+            # sd.SetSecurityDescriptorDacl(1, dacl, 0)
+            # win32security.SetFileSecurity(
+            #     testfile, win32security.DACL_SECURITY_INFORMATION, sd)
+            raise unittest.SkipTest("Requires win32api be installed")
         if os.name != "nt" and os.geteuid() == 0:
             # root ignores permission flags, so we have to skip the test
             raise unittest.SkipTest
@@ -780,36 +806,9 @@ class Test(BaseTest):
 
         file.close()
 
-        # Remove reading rights from file
+        # Remove reading rights from file. On Windows this can only set the read-only flag:
+        # https://docs.python.org/3/library/os.html#os.chmod
         os.chmod(testfile, 0o000)
-
-        if os.name == "nt":
-
-            raise unittest.SkipTest
-            # TODO: Currently skipping this test on windows as it requires `pip install win32api`
-            # which seems to have windows only dependencies.
-            # To solve this problem a requirements_windows.txt could be introduced which would
-            # then only be used on Windows.
-            #
-            # Below is some additional code to give some indication on how the implementation could
-            # look like.
-
-            from win32 import win32api
-            import win32security
-            import ntsecuritycon as con
-
-            user, domain, type = win32security.LookupAccountName(
-                "", win32api.GetUserName())
-            sd = win32security.GetFileSecurity(
-                testfile, win32security.DACL_SECURITY_INFORMATION)
-
-            dacl = win32security.ACL()
-            # Remove all access rights
-            dacl.AddAccessAllowedAce(win32security.ACL_REVISION, 0, user)
-
-            sd.SetSecurityDescriptorDacl(1, dacl, 0)
-            win32security.SetFileSecurity(
-                testfile, win32security.DACL_SECURITY_INFORMATION, sd)
 
         filebeat = self.start_beat()
 

